src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +30,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        
+        String[] newAddress = address.split(",");
```

Comment:
- An arrayIndexOutofBound error will be thrown here because all the test cases did not have all 4 component. Did run the test cases? Did you change them to enforce any assumptions you are making? If yes where for those commits?